docs/openstack/ovn_adoption.md

```diff
@@ -0,0 +1,122 @@
+    * The adoption host and the podified OVN.
+
+## Variables
+
+Define the shell variables used in the steps below. The values are
+just illustrative, use values that are correct for your environment:
+
+```bash
+OVSDB_IMAGE=quay.io/tripleozedcentos9/openstack-ovn-base:current-tripleo
+EXTERNAL_OVSDB_IP=172.17.1.49
+
+# ssh commands to reach the original controller machines
+CONTROLLER_SSH="ssh -F ~/director_standalone/vagrant_ssh_config vagrant@standalone"
+
+# ssh commands to reach the original compute machines
+COMPUTE_SSH="ssh -F ~/director_standalone/vagrant_ssh_config vagrant@standalone"
+```
+
+## Procedure
+
+- Stop OVN northd on all original cluster controllers.
+
+```bash
+${CONTROLLER_SSH} sudo systemctl stop tripleo_ovn_cluster_northd.service
+```
+
+- Backup OVN databases.
+
+```bash
+client="podman run -i --rm --userns=keep-id -u $UID -v $PWD:$PWD:z,rw -w $PWD $OVSDB_IMAGE ovsdb-client"
+${client} backup tcp:$EXTERNAL_OVSDB_IP:6641 > ovs-nb.db
+${client} backup tcp:$EXTERNAL_OVSDB_IP:6642 > ovs-sb.db
+```
+
+- Start podified OVN services prior to database import.
+
+```yaml
+oc patch openstackcontrolplane openstack --type=merge --patch '
+spec:
+  ovn:
+    enabled: true
+    template:
+      ovnDBCluster:
+        ovndbcluster-nb:
+          containerImage: quay.io/tripleozedcentos9/openstack-ovn-nb-db-server:current-tripleo
+          dbType: NB
+          storageRequest: 10G
+          networkAttachment: internalapi
+        ovndbcluster-sb:
+          containerImage: quay.io/tripleozedcentos9/openstack-ovn-sb-db-server:current-tripleo
+          dbType: SB
+          storageRequest: 10G
+          networkAttachment: internalapi
+      ovnNorthd:
+        containerImage: quay.io/tripleozedcentos9/openstack-ovn-northd:current-tripleo
+        networkAttachment: internalapi
+'
+```
+
+- Fetch podified OVN IP addresses.
+
+```bash
+PODIFIED_OVSDB_NB_IP=$(kubectl get po ovsdbserver-nb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/networks-status}' | jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | tr -d '"')
+PODIFIED_OVSDB_SB_IP=$(kubectl get po ovsdbserver-sb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/networks-status}' | jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | tr -d '"')
+```
+
+- Restore database backup to podified OVN database servers.
+
+```bash
+podman run -it --rm --userns=keep-id -u $UID -v $PWD:$PWD:z,rw -w $PWD $OVSDB_IMAGE bash -c "ovsdb-client restore tcp:$PODIFIED_OVSDB_NB_IP:6641 < ovs-nb.db"
+podman run -it --rm --userns=keep-id -u $UID -v $PWD:$PWD:z,rw -w $PWD $OVSDB_IMAGE bash -c "ovsdb-client restore tcp:$PODIFIED_OVSDB_SB_IP:6642 < ovs-sb.db"
+```
+
+- Check that podified OVN databases contain objects from backup, e.g.:
+
+```bash
+oc exec -it ovsdbserver-nb-0 -- ovn-nbctl show
+oc exec -it ovsdbserver-sb-0 -- ovn-sbctl list Chassis
+```
+
+- Switch ovn-remote on compute nodes to point to the new podified database.
+
+```bash
```

Comment:
Maybe when we add OpenStackDataPlane into the adoption instructions, this re-pointing of computes onto new OVN SB DB could be done by EDPM rather than manually. Good to have it here now though.

All LGTM, nice documentation, thanks!